src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateNow = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateNow, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateNow[key];
+        }
+        break;
+
+      case 'clear':
+        Object.keys(stateNow).forEach((key) => delete stateNow[key]);
+        break;
+
+      default:
+        break;
```

Comment:
The default case in the switch statement is empty. It's a good practice to handle unexpected 'action.type' values, perhaps by throwing an error or logging a warning.